h/static/scripts/directive/group-list.js

```diff
@@ -1,20 +1,66 @@
 'use strict';
 
+// @ngInject
+function GroupsListController($scope) {
+  $scope.expandedGroupId = undefined;
+
+  // used to keep the dropdown from closing when the user
+  // interacts with the inline link share pane within the groups list
+  $scope.stopClickPropagation = function($event) {
+    $event.stopPropagation();
+  }
+
+  // show the share link for the specified group or clear it if
+  // null
+  $scope.toggleShareLink = function (groupId) {
+    if (!groupId || $scope.expandedGroupId === groupId) {
+      $scope.expandedGroupId = undefined;
+    } else {
+      $scope.expandedGroupId = groupId;
+    }
+  };
+
+  $scope.shouldShowShareLink = function (groupId) {
+    return $scope.expandedGroupId === groupId;
+  }
+}
+
 /**
  * @ngdoc directive
  * @name groupList
  * @restrict AE
  * @description Displays a list of groups of which the user is a member.
  */
-
 // @ngInject
-module.exports = function (groups) {
+function GroupsListDirective(groups, $window) {
   return {
-    link: function (scope, elem, attrs) {
-      scope.groups = groups;
+    controller: GroupsListController,
+    link: function ($scope, elem, attrs) {
+      $scope.groups = groups;
+
+      $scope.createNewGroup = function() {
+        $window.open('/groups/new', '_blank');
+      }
+
+      $scope.$watch('expandedGroupId', function (activeGroupId) {
+        if (activeGroupId) {
+          // wait for the share link field to be revealed and then select
+          // the link's text
+          setTimeout(function() {
+            var activeShareLinkField = elem[0].querySelector('.share-link-field[data-group-id=' + activeGroupId + ']');
+            activeShareLinkField.focus();
+            activeShareLinkField.select();
+          }, 0);
+        }
+      });
     },
     restrict: 'AE',
     scope: {},
     templateUrl: 'group_list.html'
   };
 };
+
+module.exports = {
+  directive: GroupsListDirective,
+  _Controller: GroupsListController
```

Comment:
I think would consider the controller an implementation detail but the advantage of some white-box testing of units is that it makes debugging easier if something fails and the tests run faster (in this case, just some trivial function calls instead of rendering HTML in a browser), so you can have more of them.

I've added tests for the directive now which test the toggle at the component level as well.